docs/developer/languages/go.mdx

```diff
@@ -0,0 +1,481 @@
+
+* [Go](https://go.dev/doc/install) 1.23.0+
+* [TinyGo](https://tinygo.org/getting-started/install/) 0.33.0+
+* [wasmCloud Shell (`wash`)](https://wasmcloud.com/docs/installation) CLI 0.32.1+ for building and deploying components
+* [`wit-deps`](https://github.com/bytecodealliance/wit-deps) 0.3.4+ for WIT dependency management
+
+### Step 1: Set up with `wash`
+
+The wasmCloud Shell (`wash`) CLI helps developers build component-based applications that can be deployed with wasmCloud, and consolidates many open source component development tools. 
+
+Create a new project called `http-test` and navigate into the project directory:
+
+```shell
+wash new component http-test --template-name hello-world-tinygo
+```
+```shell
+cd http-test
+```
+
+Replace the contents of `deps.toml` with the WIT dependency targets below:
+
+```toml
+http = "https://github.com/WebAssembly/wasi-http/archive/v0.2.0.tar.gz"
+component = "https://github.com/wasmCloud/component-sdk-go/archive/main.tar.gz"
+```
+Run `wit-deps` from the root of your project directory:
+
+```shell
+wit-deps
+```
+
+Replace the contents of `world.wit` with the WIT definition below:
+
+```wit
+package example:http-server;
+
+world hello {
+  include wasmcloud:component/imports;
+  export wasi:http/incoming-handler@0.2.0;
+}
+```
+
+Finally, we'll delete the contents of `hello.go` and write our HTTP server, which will echo POST requests back to us.
+
+Using the Component SDK, this looks like a fairly standard server using the HTTP standard library, with only a handful of exceptions where we use methods of `wasihttp` or `wasilog`.
+
+```go
+//go:generate go run github.com/bytecodealliance/wasm-tools-go/cmd/wit-bindgen-go generate --world example --out gen ./wit
+
+package main
+
+import (
+	"io"
+	"net/http"
+	"sort"
+	"strings"
+
+	"go.wasmcloud.dev/component/log/wasilog"
+	"go.wasmcloud.dev/component/net/wasihttp"
+)
+
+const Index = `
+  /post - echo the body of a POST request`
+
+func init() {
+	wasihttp.HandleFunc(entryHandler)
+}
+
+func entryHandler(w http.ResponseWriter, r *http.Request) {
+	logger := wasilog.ContextLogger("entryHandler")
+	handlers := map[string]http.HandlerFunc{
+		"/post": postHandler,
+	}
+
+	logger.Info("Request received", "host", r.Host, "path", r.URL.Path, "agent", r.Header.Get("User-Agent"))
+
+	if handler, ok := handlers[r.URL.Path]; ok {
+		handler(w, r)
+		return
+	}
+
+	var keys []string
+	for k := range handlers {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
+	w.Header().Add("content-type", "text/plain")
+	w.Header().Add("X-Requested-Path", r.URL.Path)
+	w.Header().Add("X-Existing-Paths", strings.Join(keys, ","))
+	_, _ = w.Write([]byte(Index))
+}
+
+func postHandler(w http.ResponseWriter, r *http.Request) {
+	logger := wasilog.ContextLogger("postHandler")
+
+	w.WriteHeader(http.StatusOK)
+
+	n, err := io.Copy(w, r.Body)
+	if err != nil {
+		logger.Error("Error copying body", "error", err)
+		return
+	}
+
+	logger.Info("Copied body", "bytes", n)
+}
```

Comment:
how about we start with something much simpler, similar to:
- https://github.com/Mossaka/hello-wasi-http-tinygo
- https://github.com/sunfishcode/hello-wasi-http
- https://github.com/wasmCloud/wasmCloud/tree/8d4c48bed132fd33cc7f4d2aaa33b1999e2ea67a/examples/golang/components/http-hello-world
- https://github.com/wasmCloud/wasmCloud/blob/8d4c48bed132fd33cc7f4d2aaa33b1999e2ea67a/examples/rust/components/http-hello-world/src/lib.rs
- https://github.com/wasmCloud/wadge/blob/e5fa1f7fdb180e13252ff7a5af31da59f50fff7d/examples/go/http/http.go

such an example would make what's going on much more obvious and be much simpler to test.
Users could just `curl <url>` without having to bother with method types or body (and possibly using different HTTP clients)

```suggestion
func init() {
	wasihttp.HandleFunc(entryHandler)
}

func handler(w http.ResponseWriter, r *http.Request) {
	logger := wasilog.ContextLogger("handler")

	logger.Info("request received", "host", r.Host, "path", r.URL.Path, "agent", r.Header.Get("User-Agent"))

	_, err := io.Write(w, []byte("hello world!"))
	if err != nil {
		logger.Error("failed to write body", "error", err)
	}
}
```